components/automate-cli/pkg/diagnostics/context.go

```diff
@@ -334,15 +337,43 @@ func (c *testContext) adminToken() (string, error) {
 	for tries := 0; tries < 3; tries++ {
 		time.Sleep(time.Duration(tries) * time.Second)
 		resp, err = c.dsClient.GenerateAdminToken(context.TODO(), &api.GenerateAdminTokenRequest{
-			Name: "This token was generated by the chef-automate diagnostic tool. " +
-				"It has admin level access on the entire Automate API.",
+			Name: "admin token for diagnostic tests",
 		})
 
 		if err == nil {
-			c.Globals.CachedToken = resp.ApiToken
+			c.Globals.CachedToken = resp.TokenValue
+			c.Globals.TokenID = resp.TokenId
 			break
 		}
 	}
 
 	return c.Globals.CachedToken, nil
 }
+
+func (c *testContext) CleanupAdminToken() error {
+	resp, err := c.DoLBRequest(
+		fmt.Sprintf("/apis/iam/v2/tokens/%s", c.Globals.TokenID),
+		lbrequest.WithMethod("DELETE"),
+	)
+	if err != nil {
+		return errors.Wrap(err, "Failed to clean up the diagnostics admin token")
+	}
+	defer resp.Body.Close()
+
+	// as long as the delete was successful or the token is not found
+	// we consider the cleanup a success
+	if resp.StatusCode == 200 || resp.StatusCode == 404 {
```

Comment:
I'm not sure I like the 404 here. Besides being the case for nonexistent tokens, it's also likely what we'd get for a changed URL path.

What's the reason for keeping it? I don't suppose the token is ever not there when this runs, is it?

Reply:
yeah that's a good point. i spose if we lost the admin token at any point during the test run we would run into an error sooner, so we can assume it exists by the time we fall here.